### PR TITLE
Changes in OAuth Token struct and fixing linting issues

### PR DESCRIPTION
**Pull request by rishabh-arya95**
# Description

- Changes in the OAuth token struct
- Refactoring some test cases.
- Fixing linting issues

Fixes # (issue)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] Tested E2E using GitHub webhooks.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/core/lifecycle.go [hunk 1]
```diff
@@ -389,6 +389,6 @@ func (pl *Pipeline) getOauthSecret(repoID, gitProvider string) (*Oauth, error) {
 		pl.Logger.Errorf("error while unmarshaling json to oauth for RepoID %s : %s", repoID, err)
```
Comment (utkarsh-lambdatest) [inline pkg/core/lifecycle.go:389]: Did you do the corresponding changes in neuron token API?

Comment (rishabh-arya95) [reply pkg/core/lifecycle.go:389]: yes made the change

File: pkg/secret/secret.go [hunk 2]
```diff
@@ -54,7 +54,9 @@ func (s *secretParser) GetRepoSecret(path string) (map[string]string, error) {
 
 // GetOauthSecret parses the oauth secret
 func (s *secretParser) GetOauthSecret(path string) (*core.Oauth, error) {
-	o := &core.Oauth{}
+	o := &core.Oauth{
+		Type: core.Bearer,
```
Comment (saurabh-prakash) [inline pkg/secret/secret.go:58]: @utkarsh-lambdatest Please assess. I think this might break `Basic` auth in synapse bitbucket

Comment (utkarsh-lambdatest) [reply pkg/secret/secret.go:58]: Json unmarshal will overwrite the type, right?

File: pkg/core/lifecycle.go (hunk 1 above)
Comment (utkarsh-lambdatest) [reply pkg/core/lifecycle.go:389]: If we have removed the API in neuron, we should also remove this function and directly fetch secrets from secret parser

**Review by saurabh-prakash — APPROVED** [review]

File: pkg/secret/secret.go [hunk 3]
```diff
@@ -66,12 +68,10 @@ func (s *secretParser) GetOauthSecret(path string) (*core.Oauth, error) {
 
 	if err = json.Unmarshal(body, o); err != nil {
 		s.logger.Errorf("failed to unmarshal oauth secret, error %v", err)
-		return nil, err
+		return nil, errs.ErrUnMarshalJSON
 	}
-
-	// If tokentype is not basic set it to bearer
-	if o.Data.Type != core.Basic {
```
Comment (utkarsh-lambdatest) [inline pkg/secret/secret.go:73]: revert this as empty string token type will give error

**Review by utkarsh-lambdatest — APPROVED** [review]

**Pull request merged**